src/helperFunctions/yara_binary_search.py

```diff
@@ -51,16 +62,21 @@ def _eliminate_duplicates(result_dict):
         for key in result_dict:
             result_dict[key] = sorted(set(result_dict[key]))
 
-    def get_binary_search_result(self, yara_rules):
+    def get_binary_search_result(self, task: Tuple[bytes, Optional[str]]):
         '''
-        :param yara_rules: byte string with the contents of the yara rule file
+        :param task: tuple containing the yara_rules (byte string with the contents of the yara rule file) and optionally a firmware uid if only the contents
+                     of a single firmware are to be scanned
         :return: dict of matching rules with lists of (unique) matched UIDs as values
         '''
         with NamedTemporaryFile() as temp_rule_file:
+            yara_rules, firmware_uid = task
```

Comment:
So, as i understand it we execute yara with plain text rules, i.e. non-compiled rules. Correct?
If so, this does not introduce major overhead on a single execution, since the rules are compiled only once. If though, we run the single firmware search, we would yara possibly on thousands of files, making this thousands of compilations.

If my assumption is correct I would suggest compiling the rules into a rule file first (see yarac).